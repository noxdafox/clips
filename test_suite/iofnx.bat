(clear)                            
(open)                             ; 10.5.2.1
(open "blah1.dat")                 ; 10.5.2.1
(open "blah2.dat" blah2 "r" 10)    ; 10.5.2.1
(open 10 blah3 "r")                ; 10.5.2.1
(open [blah2.dat] blah4 "r")       ; 10.5.2.1
(open "blah4.dat" (create$) "r")   ; 10.5.2.1
(open blah5.dat blah5 r)           ; 10.5.2.1
(open blah6.dat blah6 "x")         ; 10.5.2.1
(open blah7.dat blah7 8)           ; 10.5.2.1
(open blah8.dat blah8 "r")         ; 10.5.2.1
(close blah6 89)                   ; 10.5.2.2
(close (create$))                  ; 10.5.2.2
(close [blah8])                    ; 10.5.2.2
(printout)                         ; 10.5.2.3
(printout (create$))               ; 10.5.2.3
(read (create$))                   ; 10.5.2.4
(read bogus)                       ; 10.5.2.4
(read stdin stdout)                ; 10.5.2.4
(read)                             ; 10.5.2.4 - 7
7
(read t)                           ; 10.5.2.4 - abc
abc
(read stdin)                       ; 10.5.2.4 - xyz
xyz abc
(read)                             ; 10.5.2.4 - a


    a    
(open "Temp/iofnx1.tmp" mydata "w")     ; 10.5.2.4
(printout mydata "red green")      ; 10.5.2.4
(close mydata)                     ; 10.5.2.4
(open "Temp/iofnx1.tmp" mydata)    ; 10.5.2.4
(read mydata)                      ; 10.5.2.4
(read mydata)                      ; 10.5.2.4
(read mydata)                      ; 10.5.2.4
(close mydata)                     ; 10.5.2.4
(readline (create$))               ; 10.5.2.5
(readline bogus)                   ; 10.5.2.5
(readline stdin stdout)            ; 10.5.2.5
(readline)                         ; 10.5.2.5 - "7"
7
(readline t)                       ; 10.5.2.5 - "abc"
abc
(readline stdin)                   ; 10.5.2.5 - "xyz abc"
xyz abc
(readline)                         ; 10.5.2.5 - ""

(open "Temp/iofnx1.tmp" 7.8923)    ; 10.5.2.5
(readline 7.8923)                  ; 10.5.2.5
(readline 7.8923)                  ; 10.5.2.5
(close 7.8923)                     ; 10.5.2.5
(format)                           ; 10.5.2.6
(format t)                         ; 10.5.2.6
(format (create$))                 ; 10.5.2.6
(format t "%f%%%n")                ; 10.5.2.6
(format nil "Integer: |%d|" 12)    ; 10.5.2.6
(format t "Integer: |%4d|" 12)     ; 10.5.2.6
(format nil "Integer: |%-04d|" 12)
(format t "Float:   |%f|" 12.01)   ; 10.5.2.6
(format nil "Float:   |%7.2f| "12.01)
(format t "Test:    |%e|" 12.01)   ; 10.5.2.6
(format nil "Test:    |%7.2e|" 12.01)
(format t "General: |%g|" 1234567890)
(format t "Hexadecimal: |%x|" 12)  ; 10.5.2.6
(format t "Octal:   |%o|" 12)      ; 10.5.2.6
(format nil "Symbols: |%s| |%s|" value-a1 capacity)
(format nil "the %% x %d x %s x %f y %%" 4 ab 3.5)
(format nil "%d" abc)              ; 10.5.2.6
(format nil "%d" 9.8)              ; 10.5.2.6
(format t "%f" 40)                 ; 10.5.2.6
(format nil "%f" abc)              ; 10.5.2.6
(format nil "%g" (create$))        ; 10.5.2.6
(format t "%o" 9.8)                ; 10.5.2.6
(remove "Temp/iofun.dat")          ; Open function modes
(open "Temp/iofun.dat" temp "bogus")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "r")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "w")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "a")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "rb")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "wb")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "ab")
(close temp)
(remove "Temp/iofun.dat") 
(open "Temp/iofun.dat" temp "w")
(printout temp red crlf)
(close temp)
(open "Temp/iofun.dat" temp "r")
(read temp)
(read temp)
(close temp)
(open "Temp/iofun.dat" temp "a")
(read temp)
(printout temp green crlf)
(close temp)
(open "Temp/iofun.dat" temp "r")
(read temp)
(read temp)
(read temp)
(close temp)
(remove "Temp/iofun.dat") 
