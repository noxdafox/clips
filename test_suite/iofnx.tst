(unwatch all)
(clear)
(set-strategy depth)
(open "Results//iofnx.rsl" iofnx "w")
(dribble-on "Actual//iofnx.out")
(batch "iofnx.bat")
(dribble-off)
(load "compline.clp")
(printout iofnx "iofnx.bat differences are as follows:" crlf)
(compare-files "Expected//iofnx.out" "Actual//iofnx.out" iofnx)
; close result file
(close iofnx)
