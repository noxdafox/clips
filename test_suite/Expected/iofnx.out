TRUE
CLIPS> (batch "iofnx.bat")
TRUE
CLIPS> (clear)                            
CLIPS> (open)                             ; 10.5.2.1
[ARGACCES4] Function open expected at least 2 argument(s)
CLIPS> (open "blah1.dat")                 ; 10.5.2.1
[ARGACCES4] Function open expected at least 2 argument(s)
CLIPS> (open "blah2.dat" blah2 "r" 10)    ; 10.5.2.1
[ARGACCES4] Function open expected no more than 3 argument(s)
CLIPS> (open 10 blah3 "r")                ; 10.5.2.1
[ARGACCES5] Function open expected argument #1 to be of type symbol or string
CLIPS> (open [blah2.dat] blah4 "r")       ; 10.5.2.1
[ARGACCES5] Function open expected argument #1 to be of type symbol or string
CLIPS> (open "blah4.dat" (create$) "r")   ; 10.5.2.1
[IOFUN1] Illegal logical name used for open function.
FALSE
CLIPS> (open blah5.dat blah5 r)           ; 10.5.2.1
[ARGACCES5] Function open expected argument #3 to be of type string
FALSE
CLIPS> (open blah6.dat blah6 "x")         ; 10.5.2.1
[ARGACCES5] Function open expected argument #3 to be of type string with value "r", "w", "a", "rb", "wb", or "ab"
FALSE
CLIPS> (open blah7.dat blah7 8)           ; 10.5.2.1
[ARGACCES5] Function open expected argument #3 to be of type string
FALSE
CLIPS> (open blah8.dat blah8 "r")         ; 10.5.2.1
FALSE
CLIPS> (close blah6 89)                   ; 10.5.2.2
[ARGACCES4] Function close expected no more than 1 argument(s)
CLIPS> (close (create$))                  ; 10.5.2.2
[IOFUN1] Illegal logical name used for close function.
FALSE
CLIPS> (close [blah8])                    ; 10.5.2.2
FALSE
CLIPS> (printout)                         ; 10.5.2.3
[ARGACCES4] Function printout expected at least 1 argument(s)
CLIPS> (printout (create$))               ; 10.5.2.3
[IOFUN1] Illegal logical name used for printout function.
CLIPS> (read (create$))                   ; 10.5.2.4
[IOFUN1] Illegal logical name used for read function.
"*** READ ERROR ***"
CLIPS> (read bogus)                       ; 10.5.2.4
[ROUTER1] Logical name bogus was not recognized by any routers
"*** READ ERROR ***"
CLIPS> (read stdin stdout)                ; 10.5.2.4
[ARGACCES4] Function read expected no more than 1 argument(s)
CLIPS> (read)                             ; 10.5.2.4 - 7
7
7
CLIPS> (read t)                           ; 10.5.2.4 - abc
abc
abc
CLIPS> (read stdin)                       ; 10.5.2.4 - xyz
xyz abc
xyz
CLIPS> (read)                             ; 10.5.2.4 - a


    a    
a
CLIPS> (open "Temp/iofnx1.tmp" mydata "w")     ; 10.5.2.4
TRUE
CLIPS> (printout mydata "red green")      ; 10.5.2.4
CLIPS> (close mydata)                     ; 10.5.2.4
TRUE
CLIPS> (open "Temp/iofnx1.tmp" mydata)    ; 10.5.2.4
TRUE
CLIPS> (read mydata)                      ; 10.5.2.4
red
CLIPS> (read mydata)                      ; 10.5.2.4
green
CLIPS> (read mydata)                      ; 10.5.2.4
EOF
CLIPS> (close mydata)                     ; 10.5.2.4
TRUE
CLIPS> (readline (create$))               ; 10.5.2.5
[IOFUN1] Illegal logical name used for readline function.
"*** READ ERROR ***"
CLIPS> (readline bogus)                   ; 10.5.2.5
[ROUTER1] Logical name bogus was not recognized by any routers
"*** READ ERROR ***"
CLIPS> (readline stdin stdout)            ; 10.5.2.5
[ARGACCES4] Function readline expected no more than 1 argument(s)
CLIPS> (readline)                         ; 10.5.2.5 - "7"
7
"7"
CLIPS> (readline t)                       ; 10.5.2.5 - "abc"
abc
"abc"
CLIPS> (readline stdin)                   ; 10.5.2.5 - "xyz abc"
xyz abc
"xyz abc"
CLIPS> (readline)                         ; 10.5.2.5 - ""

""
CLIPS> (open "Temp/iofnx1.tmp" 7.8923)    ; 10.5.2.5
TRUE
CLIPS> (readline 7.8923)                  ; 10.5.2.5
"red green"
CLIPS> (readline 7.8923)                  ; 10.5.2.5
EOF
CLIPS> (close 7.8923)                     ; 10.5.2.5
TRUE
CLIPS> (format)                           ; 10.5.2.6
[ARGACCES4] Function format expected at least 2 argument(s)
CLIPS> (format t)                         ; 10.5.2.6
[ARGACCES4] Function format expected at least 2 argument(s)
CLIPS> (format (create$))                 ; 10.5.2.6
[ARGACCES4] Function format expected at least 2 argument(s)
CLIPS> (format t "%f%%%n")                ; 10.5.2.6
[ARGACCES4] Function format expected exactly 3 argument(s)
""
CLIPS> (format nil "Integer: |%d|" 12)    ; 10.5.2.6
"Integer: |12|"
CLIPS> (format t "Integer: |%4d|" 12)     ; 10.5.2.6
Integer: |  12|"Integer: |  12|"
CLIPS> (format nil "Integer: |%-04d|" 12)
"Integer: |12  |"
CLIPS> (format t "Float:   |%f|" 12.01)   ; 10.5.2.6
Float:   |12.010000|"Float:   |12.010000|"
CLIPS> (format nil "Float:   |%7.2f| "12.01)
"Float:   |  12.01| "
CLIPS> (format t "Test:    |%e|" 12.01)   ; 10.5.2.6
Test:    |1.201000e+01|"Test:    |1.201000e+01|"
CLIPS> (format nil "Test:    |%7.2e|" 12.01)
"Test:    |1.20e+01|"
CLIPS> (format t "General: |%g|" 1234567890)
General: |1.23457e+09|"General: |1.23457e+09|"
CLIPS> (format t "Hexadecimal: |%x|" 12)  ; 10.5.2.6
Hexadecimal: |c|"Hexadecimal: |c|"
CLIPS> (format t "Octal:   |%o|" 12)      ; 10.5.2.6
Octal:   |14|"Octal:   |14|"
CLIPS> (format nil "Symbols: |%s| |%s|" value-a1 capacity)
"Symbols: |value-a1| |capacity|"
CLIPS> (format nil "the %% x %d x %s x %f y %%" 4 ab 3.5)
"the % x 4 x ab x 3.500000 y %"
CLIPS> (format nil "%d" abc)              ; 10.5.2.6
[ARGACCES5] Function format expected argument #3 to be of type integer or float
""
CLIPS> (format nil "%d" 9.8)              ; 10.5.2.6
"9"
CLIPS> (format t "%f" 40)                 ; 10.5.2.6
40.000000"40.000000"
CLIPS> (format nil "%f" abc)              ; 10.5.2.6
[ARGACCES5] Function format expected argument #3 to be of type integer or float
""
CLIPS> (format nil "%g" (create$))        ; 10.5.2.6
[ARGACCES5] Function format expected argument #3 to be of type integer or float
""
CLIPS> (format t "%o" 9.8)                ; 10.5.2.6
11"11"
CLIPS> (remove "Temp/iofun.dat")          ; Open function modes
FALSE
CLIPS> (open "Temp/iofun.dat" temp "bogus")
[ARGACCES5] Function open expected argument #3 to be of type string with value "r", "w", "a", "rb", "wb", or "ab"
FALSE
CLIPS> (close temp)
FALSE
CLIPS> (remove "Temp/iofun.dat") 
FALSE
CLIPS> (open "Temp/iofun.dat" temp "r")
FALSE
CLIPS> (close temp)
FALSE
CLIPS> (remove "Temp/iofun.dat") 
FALSE
CLIPS> (open "Temp/iofun.dat" temp "w")
TRUE
CLIPS> (close temp)
TRUE
CLIPS> (remove "Temp/iofun.dat") 
TRUE
CLIPS> (open "Temp/iofun.dat" temp "a")
TRUE
CLIPS> (close temp)
TRUE
CLIPS> (remove "Temp/iofun.dat") 
TRUE
CLIPS> (open "Temp/iofun.dat" temp "rb")
FALSE
CLIPS> (close temp)
FALSE
CLIPS> (remove "Temp/iofun.dat") 
FALSE
CLIPS> (open "Temp/iofun.dat" temp "wb")
TRUE
CLIPS> (close temp)
TRUE
CLIPS> (remove "Temp/iofun.dat") 
TRUE
CLIPS> (open "Temp/iofun.dat" temp "ab")
TRUE
CLIPS> (close temp)
TRUE
CLIPS> (remove "Temp/iofun.dat") 
TRUE
CLIPS> (open "Temp/iofun.dat" temp "w")
TRUE
CLIPS> (printout temp red crlf)
CLIPS> (close temp)
TRUE
CLIPS> (open "Temp/iofun.dat" temp "r")
TRUE
CLIPS> (read temp)
red
CLIPS> (read temp)
EOF
CLIPS> (close temp)
TRUE
CLIPS> (open "Temp/iofun.dat" temp "a")
TRUE
CLIPS> (read temp)
EOF
CLIPS> (printout temp green crlf)
CLIPS> (close temp)
TRUE
CLIPS> (open "Temp/iofun.dat" temp "r")
TRUE
CLIPS> (read temp)
red
CLIPS> (read temp)
green
CLIPS> (read temp)
EOF
CLIPS> (close temp)
TRUE
CLIPS> (remove "Temp/iofun.dat") 
TRUE
CLIPS> (dribble-off)
